be/src/common/config.h

```diff
@@ -838,6 +838,9 @@ CONF_Int32(jdbc_connection_idle_timeout_ms, "600000");
 // spill dirs
 CONF_String(spill_local_storage_dir, "spill");
 CONF_mBool(experimental_spill_skip_sync, "false");
+// The maximum size of a single log block container file, this is not a hard limit.
+// If the file size exceeds this limit, a new file will be created to store the block.
+CONF_Int64(experimental_spill_max_log_block_container_bytes, "10737418240"); // 10GB
```

Comment:
1. should we preserve or control the total size of block_container to reduce memory useage?
2. the size of block_containers is limited by mem_tracker?

Reply:
> 1. should we preserve or control the total size of block_container to reduce memory useage?
> 2. the size of block_containers is limited by mem_tracker?

the total size of a block container depends on how many Block are inside, which is related to the size of Block. It's hard to control precisely. We have MemGuard to track memory usage during restore and flush, so the memory usage of block containers will be limited by mem_tracker

Reply:
we should limit the memory usage of spill write buff and read buff to avoid they taking too much memory. for example, spill used buffers take 10%, executor takes 80%, others takes 10%. if exceeding the limit and the executor's limit is full, the spill write buff should flush data to disk.

Reply:
> we should limit the memory usage of spill write buff and read buff to avoid they taking too much memory. for example, spill used buffers take 10%, executor takes 80%, others takes 10%. if exceeding the limit and the executor's limit is full, the spill write buff should flush data to disk.

I got your point, I will think about it and optimize later.